gridcomps/History/MAPL_HistoryTrajectoryMod_smod.F90

```diff
@@ -262,404 +392,304 @@
-         
-         _RETURN(_SUCCESS)
+         real(ESMF_KIND_R8), pointer :: ptAT(:)
+         type(ESMF_routehandle) :: RH
+         type(ESMF_Time) :: timeset(2)
+         type(ESMF_Time) :: current_time
+         type(ESMF_Time) :: time0, time1, time2
+         type(ESMF_TimeInterval) :: interval
+         type(ESMF_Grid) :: grid
 
-       end procedure append_file
+         type(ESMF_VM) :: vm
+         integer :: mypet, petcount
 
+         integer :: i, j, k, L
+         integer :: fid_s, fid_e
+         integer(kind=ESMF_KIND_I8) :: j0, j1
+         integer(kind=ESMF_KIND_I8) :: jt1, jt2
+         integer(kind=ESMF_KIND_I8) :: nstart, nend
+         real(kind=ESMF_KIND_R8) :: jx0, jx1
+         integer :: nx, nx_sum
+         integer :: arr(1)
+         integer :: sec
+         integer :: int_time
+         character(len=:), allocatable :: tunit
+         integer (ESMF_KIND_I8) :: n_second
+         integer, allocatable :: ix(:) !  counter for each obs(k)%nobs_epoch
+         integer :: nx2
 
-      module procedure reset_times_to_current_day
 
-         integer :: i,status,h,m,yp,mp,dp,s,ms,us,ns
-         type(ESMF_Clock) :: clock
-         type(ESMF_Time) :: current_time
-         integer :: year,month,day
+         this%datetime_units = "seconds since 1970-01-01 00:00:00"
+         lgr => logging%get_logger('HISTORY.sampler')
 
-         call this%time_info%get(clock=clock,_RC)
-         call ESMF_ClockGet(clock,currtime=current_time,_RC)
-         call ESMF_TimeGet(current_time,yy=year,mm=month,dd=day,_RC)
-         do i=1,size(this%times)
-            call ESMF_TimeGet(this%times(i),yy=yp,mm=mp,dd=dp,h=h,m=m,s=s,ms=ms,us=us,ns=ns,_RC)
-            call ESMF_TimeSet(this%times(i),yy=year,mm=month,dd=day,h=h,m=m,s=s,ms=ms,us=us,ns=ns,_RC)
-         enddo
+         call ESMF_VMGetGlobal(vm,_RC)
+         call ESMF_VMGet(vm, localPet=mypet, petCount=petCount, _RC)
 
-      end procedure reset_times_to_current_day
+         if (this%nc_index == '') then
+            !
+            !-- non IODA case
+            !
+            _FAIL('non-IODA format is not implemented here')
+            !! sorry Ben, cannot make everything work
+            !!filename=trim(this%obsFile)
+            !!!!         print*, 'obs file name=', trim(filename)
+            !!call formatter%open(trim(filename),pFIO_READ,_RC)
+            !!fmd = formatter%read(_RC)
+            !!call metadata_utils%create(fmd,trim(filename))
+            !!num_times = metadata_utils%get_dimension("time",_RC)
+            !!allocate(this%lons(num_times),this%lats(num_times),_STAT)
+            !!if (metadata_utils%is_var_present("longitude")) then
+            !!   call formatter%get_var("longitude",this%lons,_RC)
+            !!end if
+            !!if (metadata_utils%is_var_present("latitude")) then
+            !!   call formatter%get_var("latitude",this%lats,_RC)
+            !!end if
+            !!call metadata_utils%get_time_info(timeVector=this%times,_RC)
```

Comment:
A reminder to clean up commented code.    Can be deferred until after this is merged.

Reply:
I will clean it up and update the PR.